cvat-sdk/cvat_sdk/core/downloading.py

```diff
@@ -34,14 +34,17 @@ def download_file(
         *,
         timeout: int = 60,
         pbar: Optional[ProgressReporter] = None,
-    ) -> None:
+    ) -> Path:
         """
         Downloads the file from url into a temporary file, then renames it to the requested name.
+        If output_path is a directory, saves the file into the directory with
+        the server-defined name.
         """
 
         CHUNK_SIZE = 10 * 2**20
 
-        assert not output_path.exists()
+        if output_path.exists() and not output_path.is_dir():
+            raise FileExistsError(output_path)
```

Comment:
Perhaps this check should be done later, after the server-defined name has been received?

Reply:
Agree, added a check for this case.